remix/app1/webpack.server.js

```diff
@@ -0,0 +1,171 @@
+}
+
+const manifest = getManifest();
+const serverBuildModule = "./.cache/server-build.js";
+const serverBuildEntry = createServerBuildEntry(remixConfig, manifest);
+fs.writeFileSync(serverBuildModule, serverBuildEntry, "utf8");
+
+
+/**
+ * @type {import('webpack').Configuration}
+ */
+const config = {
+  name: "server",
+  mode,
+  devtool: mode === "development" ? false : undefined,
+  target: 'async-node',
+  entry: remixConfig.serverEntryPoint
+    ? path.resolve(remixConfig.rootDirectory, remixConfig.serverEntryPoint)
+    : serverBuildModule,
+  experiments: isModule ? {outputModule: true} : undefined,
+  externalsType: isModule ? "module" : undefined,
+  externalsPresets: {node: true},
+  externals: [
+    nodeExternals({
+      allowlist: [/^@remix-run\/dev/],
+      importType: isModule ? "module" : "commonjs",
+    }),
+  ],
+  output: {
+    environment: {
+      module: isModule
+    },
+    filename: path.basename(remixConfig.serverBuildPath),
+    library: {type: isModule ? "module" : "commonjs"},
+    chunkFormat: isModule ? "module" : "commonjs",
+    chunkLoading: isModule ? "import" : undefined,
+    module: isModule,
+    path: path.dirname(remixConfig.serverBuildPath),
+    publicPath: remixConfig.publicPath,
+    assetModuleFilename: "_assets/[name]-[contenthash][ext]",
+    cssChunkFilename: "_assets/[name]-[contenthash][ext]",
+    chunkFilename: "[name]-[chunkhash].js",
+  },
+  optimization: {
+    moduleIds: "named",
+  },
+  resolve: {
+    alias: {
+      "@remix-run/dev/server-build.js": serverBuildModule,
+    },
+  },
+  module: {
+    rules: [
+      {
+        test: /\.[jt]sx?$/,
+        use: [
+          {
+            loader: "esbuild-loader",
+            options: {
+              target: "es2019",
+              jsx: "automatic",
+            },
+          },
+        ],
+      },
+    ],
+  },
+  plugins: [
+    new AsyncBoundaryPlugin({
+      excludeChunk: (chunk) => {
+        console.log(chunk.name)
+        return chunk.name === 'app1'
+      }
+    }),
+    new UniversalFederationPlugin({
+      isServer: true,
+      name: "app1",
+      filename: 'remoteEntry.js',
+      remoteType: 'script',
+      remotes: {
+        app2: 'app2@http://localhost:3001/server/remoteEntry.js'
+      },
+      library: {type: isModule ? "module" : 'commonjs-module'},
+      exposes: {
+        './button': './components/Button.jsx',
+      },
+      shared: {
+        "react/": {
+          singleton: true
+        },
+        "react": {
+          singleton: true
+        },
+        "react-dom/": {
+          singleton: true
+        },
+        "react-dom": {
+          singleton: true
+        }
+      }
+    }, {ModuleFederationPlugin}),
+  ]
+};
+
+export default config;
+
+/**
+ *
+ * @param {import("@remix-run/dev").ResolvedRemixConfig} config
+ * @param {import("@remix-run/dev").AssetsManifest} manifest
+ * @returns
+ */
+function createServerBuildEntry(config, manifest) {
```

Comment:
Attach Module Federation v1.5 hooks here, perform distributed route discovery across remotes automatically. 
Easy to do, next.js made it impossible